src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -867,19 +867,11 @@ private boolean canBeReferencedFromStaticContext(DetailAST ident) {
             staticContext = true;
         }
         else {
-            if (variableDeclarationFrame.getType() == FrameType.CLASS_FRAME) {
-                final DetailAST codeBlockDefinition = getCodeBlockDefinitionToken(ident);
-                if (codeBlockDefinition != null) {
-                    final DetailAST modifiers = codeBlockDefinition.getFirstChild();
-                    staticContext = codeBlockDefinition.getType() == TokenTypes.STATIC_INIT
-                        || modifiers.findFirstToken(TokenTypes.LITERAL_STATIC) != null;
-                }
-            }
-            else {
-                final DetailAST frameNameIdent = variableDeclarationFrame.getFrameNameIdent();
-                final DetailAST definitionToken = frameNameIdent.getParent();
-                staticContext = definitionToken.findFirstToken(TokenTypes.MODIFIERS)
-                        .findFirstToken(TokenTypes.LITERAL_STATIC) != null;
+            final DetailAST codeBlockDefinition = getCodeBlockDefinitionToken(ident);
+            if (codeBlockDefinition != null) {
+                final DetailAST modifiers = codeBlockDefinition.getFirstChild();
+                staticContext = codeBlockDefinition.getType() == TokenTypes.STATIC_INIT
+                    || modifiers.findFirstToken(TokenTypes.LITERAL_STATIC) != null;
```

Comment:
Actually, this condition is sufficient for the whole method, but as we are killing mutations one by one, let's wait for https://github.com/checkstyle/checkstyle/pull/11792 to get merged, after which we can modify the complete method.

Reply:
https://github.com/checkstyle/checkstyle/pull/11792 is merged, please rebase and update this PR.